x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/allocation/SearchableSnapshotDiskThresholdIntegTests.java

```diff
@@ -0,0 +1,287 @@
+                    false,
+                    storage
+                ),
+                ActionListener.wrap(response -> mountLatch.countDown(), e -> mountLatch.countDown())
+            );
+        }
+        mountLatch.await();
+
+        // The cold/frozen data node has enough disk space to hold all the shards
+        assertBusy(() -> {
+            var state = client().admin().cluster().prepareState().setRoutingTable(true).get().getState();
+            assertThat(
+                state.routingTable()
+                    .allShards()
+                    .stream()
+                    .filter(shardRouting -> state.metadata().index(shardRouting.shardId().getIndex()).isSearchableSnapshot())
+                    .allMatch(
+                        shardRouting -> shardRouting.state() == ShardRoutingState.STARTED
+                            && otherDataNodeId.equals(shardRouting.currentNodeId())
+                    ),
+                equalTo(true)
+            );
+        });
+
+        final CountDownLatch extraLatch = new CountDownLatch(indicesStoresSizes.size());
+        final String extraPrefix = "extra-";
+
+        logger.info("--> mounting [{}] indices with [{}] prefix", indicesStoresSizes.size(), extraPrefix);
+        for (String index : indicesStoresSizes.keySet()) {
+            client().execute(
+                MountSearchableSnapshotAction.INSTANCE,
+                new MountSearchableSnapshotRequest(
+                    extraPrefix + index,
+                    repository,
+                    snapshot,
+                    randomFrom(indicesStoresSizes.keySet()),
+                    Settings.EMPTY,
+                    Strings.EMPTY_ARRAY,
+                    false,
+                    storage
+                ),
+                ActionListener.wrap(response -> extraLatch.countDown(), e -> extraLatch.countDown())
+            );
+        }
+        extraLatch.await();
+
+        // TODO Indices should not be allocated without checking the node disk usage first
+        assertBusy(() -> {
+            var state = client().admin().cluster().prepareState().setRoutingTable(true).get().getState();
+            assertThat(
+                state.routingTable()
+                    .allShards()
+                    .stream()
+                    .filter(
+                        shardRouting -> shardRouting.shardId().getIndexName().startsWith(extraPrefix)
+                            && state.metadata().index(shardRouting.shardId().getIndex()).isSearchableSnapshot()
+                    )
+                    .allMatch(
+                        shardRouting -> shardRouting.state() == ShardRoutingState.STARTED
+                            && otherDataNodeId.equals(shardRouting.currentNodeId())
+                    ),
+                equalTo(true)
+            );
+        });
+    }
+
+    private static Map<String, Long> sizeOfShardsStores(String indexPattern) {
+        final Map<String, Long> sizes = new HashMap<>();
```

Comment:
WDYT about expressing the computation with the Streams API?

```java
Arrays.stream(client().admin().indices().prepareStats(indexPattern).clear().setStore(true).get().getShards())
      .collect(Collectors.toUnmodifiableMap(s -> s.getShardRouting().getIndexName(), s -> s.getStats().getStore().sizeInBytes(), Long::sum))
```

Reply:
Sure